servant-docs/src/Servant/Docs/Internal.hs

```diff
@@ -161,6 +161,20 @@ data DocQueryParam = DocQueryParam
   , _paramKind   :: ParamKind
   } deriving (Eq, Ord, Show)
 
+-- | A type to represent fragment. Holds the name of the fragment and its description.
+--
+-- Write a 'ToFragment' instance for your fragment types.
+data DocFragment = DocFragment
+  { _fragSymbol :: String -- type supplied
+  , _fragDesc   :: String -- user supplied
+  } deriving (Eq, Ord, Show)
+
+-- | There should be at most one `Fragment` per API endpoint.
+-- So here we are keeping the first occurrence.
+combineFragment :: Maybe DocFragment -> Maybe DocFragment -> Maybe DocFragment
+Nothing `combineFragment` mdocFragment = mdocFragment
```

Comment:
This is equal (i think) to `combineFragment a b = getFirst $ (First a) <> (First b)`

It's up to you to decide if you want to make this change.

Reply:
I left this change as it was before. 
If you insist, I could definitely change it.